### PR TITLE
fix saturation list creation

### LINKED ISSUE #41: enhancement/fix needed in saturated_sites.csv
I am going through 4.image-and-segmentation.py in much more detail in #39 - the following code block is giving me some trouble:

https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/blob/b8c9288c0ab8b04e4ce284be5ba9aab44a6ce678/0.preprocess-sites/4.image-and-segmentation-qc.py#L345-L357

I am trying to think through what this is doing (so please bear with me!)

There are two different `ImageQuality_PercentMaximal` types (spots and cells) and this column, in `image.csv` indicates whether or not that image (either spots or cells) is over saturated? We also want these sites to be output to a file named `saturated_sites.csv` but _only if any saturated sites exist_. Is this correct?

If so, then I think we may need to slightly tinker the code to make sure we extract all saturated sites. Sorry I didn't flag this sooner!

### PR DESCRIPTION
**Pull request by ErinWeisbart**
closes #41

### CODE REVIEW

File: 0.preprocess-sites/4.image-and-segmentation-qc.py [hunk 1]
```diff
@@ -317,32 +317,59 @@
...
-        if check_if_write(output_file, force, throw_warning=True):
-            sat_df.to_csv(sat_output_file)
+        temp = image_df[image_df[col] > 0.25]
+        bc_sat_df = bc_sat_df.append(temp)
+    if len(bc_sat_df) != 0:
+        bc_sat_df.loc[:, "Fails_BC_Sat"] = "Fails"
+
+if len(cp_sat_df) > 0 and len(bc_sat_df) > 0:
+    sat_df = (
+        cp_sat_df.set_index("Metadata_Site_Full")
```
Comment (gwaygenomics) [inline 0.preprocess-sites/4.image-and-segmentation-qc.py:353]: is `Metadata_Site_Full` defined somewhere else as an important column? What would happen if this column isn't present?

File: 0.preprocess-sites/4.image-and-segmentation-qc.py [hunk 2]
```diff
@@ -317,32 +317,59 @@
...
+        .combine_first(bc_sat_df.set_index("Metadata_Site_Full"))
+        .reset_index()
+    )
+elif len(cp_sat_df) > 0:
+    sat_df = cp_sat_df
+elif len(bc_sat_df) > 0:
+    sat_df = bc_sat_df
```
[the remark this answers is not in the document]
Comment (ErinWeisbart) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py:360]: We can thank StackOverflow for my (and now your) introduction to it :)

File: 0.preprocess-sites/4.image-and-segmentation-qc.py [hunk 3]
```diff
@@ -317,32 +317,59 @@
...
+
+if not sat_df.empty:
+    sat_df_cols = cp_sat_cols + bc_sat_cols + addn_cols
```
[the remark this answers is not in the document]
Comment (ErinWeisbart) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py:361]: `Metadata_Site_Full` comes from the cell_count.tsv made in 0./3.visualize-cell-summary where it is saved from the cell_count_df. 
0./3. cell_count_df is made from the individual cell_counts_{site}.tsv made in 0./2.process-cells. 
That file is made from the cell_count_df in 0./2. where the creation of `Metadata_Site_Full` is hardcoded.
So, I think that means we don't need to check for `Metadata_Site_Full`?

I've fixed the addition of `Fails_CP_Sat` and `Fails_BC_Sat`

File: 0.preprocess-sites/4.image-and-segmentation-qc.py (hunk 1 above)
Comment (ErinWeisbart) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py]: See comment above for where `Metadata_Site_Full` is hardcoded. Therefore okay to hardcode here?

**Comment by ErinWeisbart** [comment]
@gwaygenomics what do you think now?
At this point can we assume that if we've hardcoded something in an earlier step (e.g. `Metadata_Site_Full`) that it's okay to hardcode it at a later step?

File: 0.preprocess-sites/4.image-and-segmentation-qc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gwaygenomics) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py:365]: Wait, doesn't the value come from https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/blob/master/0.preprocess-sites/1.process-spots.py#L113 where we assign`Metadata_Site_Full` as a column to the image_df?

I think that #45 is going to impact this as well. We should decide something quickly in that issue. (about what nomenclature to stick with!)

File: 0.preprocess-sites/4.image-and-segmentation-qc.py (hunk 1 above)
Comment (gwaygenomics) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py:352]: i think so, but we should try to be internally consistent with nomenclature moving forward. This is my bad! We should have thought through this decision in more detail together - would have saved some time

**Comment by gwaygenomics** [comment]
Looks good! Let's resolve #45 first before merging this

File: 0.preprocess-sites/4.image-and-segmentation-qc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErinWeisbart) [reply 0.preprocess-sites/4.image-and-segmentation-qc.py:365]: Yes! Sorry, I had written 0./2. where it should have said 0./1. (it's fixed in comment now). Your link is the correct location.

**Comment by gwaygenomics** [comment]
ok! I think this is ready to merge. I still need to work on #43 - and specifically this file 😱 . If we merge this in now, then I can work on a more final version of this script, without having to worry about possibly stepping too much on these changes.

Please merge when you think the PR is ready/ when you get a chance 🎉

**Pull request merged**